### ISSUE TITLE
Avoid httpredir.debian.net

### ISSUE THREAD

**Issue by lifeofguenter** [andsens/bootstrap-vz#338]
Thanks for creating this awesome project!

I would like to propose for the replacement of the debian mirror (even though its just a redirector) httpredir.debian.net.

Yes I know it is only the default, thus replaceable, but it causes a newly launched EC2 instance with Debian to be randomly unstable at times.
- many of my tests concluded that the mirror selected will not even be in the same country, even though that country has plenty of mirrors
- ec2 instances should strictly use https://cloudfront.debian.net/ for highest availability and lowest latency
- randomly it will select a mirror that is defect, causing automated scripts to fail

In short: httpredir.debian.net is great for private use, but not when every update has to be a non-fail - e.g. when HA/reliability is important.

**Comment by nbraud** [comment]
Pitching in here: I have been having trouble too with `httpredir.debian.org`, and it hasn't been very reliable.
Moreover, I learnt during DebConf that it is currently unmaintained.

I would like to suggest:
- [ ] switching to `deb.debian.org` as the default mirror (and default security mirror), as it is backed by a CDN and maintained by DSA;
- [ ] making the default mirror, in the special case of the `ec2` provider, be `cloudfront.debian.net` (but not as a security mirror, it is not yet maintained by DSA).

**Comment by andsens** [comment]
👍  on both @nbraud. I don't have a lot of time right now. But I'll merge if someone sends a PR.

**Comment by vorlock** [comment]
@andsens I think this can be closed

**Comment by andsens** [comment]
Agreed. Closing.

**Comment by nbraud** [comment]
Quickly asked jcristau on IRC, and AWS clients don't see the `deb.d.o` record for cloudfront with a higher weight (also, they pay the penalty of doing the HTTP request to `deb.debian.org` before getting a redirect, pre-Stretch), so I guess there is still a reason to implement the different default for EC2 images.
